scripts/ci/check_compliance.py

```diff
@@ -212,7 +212,7 @@ def run(self):
         if not os.path.exists(checkpatch):
             self.skip(f'{checkpatch} not found')
 
-        diff = subprocess.Popen(('git', 'diff', '--no-ext-diff', COMMIT_RANGE),
+        diff = subprocess.Popen(('git', 'diff', '--no-ext-diff', '--', ':!*.diff', ':!*.patch', COMMIT_RANGE),
```

Comment:
Why do you consider these to be false positive? Where are the patch/diff files coming from?

Reply:
If the original source file has whitespace errors, then a diff or patch against the original source file can also have whitespace errors. Is the diff or patch file in violation? Not really.

And in fact, when building against a specific commit of Zephyr there is no way to simultaneously keep the patch or diff minimal while also ensuring that whitespace errors